### PR TITLE
cr: only resurrect deleted, unmerged rm ops

### PR DESCRIPTION
**Pull request by strib**
If a directory was modified and then deleted on the unmerged branch, but modified on the merged branch, we only need to replay the unmerged "remove" operations.  Anything else like setAttrs and syncs are irrelevant, since the nodes were deleted.

In the course of debugging I factored out some logic into its own function.  I didn't end up needing to do that, but I kept it anyway as a nice cleanup.  It's unrelated to the rest of this PR.

Issue: KBFS-1691
Issue: keybase/client#9478

### CODE REVIEW

**Comment by songgao** [comment]
Btw your branch name suggests https://keybase.atlassian.net/browse/KBFS-2507 . Is that related?

**Comment by strib** [comment]
> Btw your branch name suggests https://keybase.atlassian.net/browse/KBFS-2507 . Is that related?

I thought it was, it's a similar crash, but on second look it's in a different, unrelated place.  I just didn't feel like changing the branch name and having to open a new PR.

File: libkbfs/conflict_resolver.go [hunk 1]
```diff
@@ -981,6 +981,16 @@ func (cr *ConflictResolver) resolveMergedPaths(ctx context.Context,
 			"also modified in the merged branch %v",
 			unmergedChain.mostRecent, mergedChain.mostRecent)
 
+		// We know that everything in the directory has been removed,
+		// so only rm ops matter.
+		var newOps []op
```
Comment (jzila) [inline libkbfs/conflict_resolver.go]: s/`newOps`/`rmOps`

**Comment by strib** [comment]
User confirmed this fixed their issue.

**Pull request merged**